tests/shell/tests.py

```diff
@@ -94,3 +95,30 @@ def test_shell_with_bpython_not_installed(self, select):
     # returns EOF and so select always shows that sys.stdin is ready to read.
     # This causes problems because of the call to select.select() toward the
     # end of shell's handle() method.
+
+    def test_ipython(self):
+        cmd = shell.Command()
+        mock_ipython = mock.Mock(start_ipython=mock.MagicMock())
+
+        with mock.patch.dict(sys.modules, {"IPython": mock_ipython}):
+            cmd.ipython(None)
```

Comment:
Use an empty dict, which at least has the right type for `options`:

```suggestion
            cmd.ipython({})
```